substrate/frame/ranked-collective/src/lib.rs

```diff
@@ -840,6 +848,91 @@ pub mod pallet {
 		}
 	}
 
+	#[cfg(any(feature = "try-runtime", test))]
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		/// Ensure the correctness of the state of this pallet.
+		pub fn do_try_state() -> Result<(), sp_runtime::TryRuntimeError> {
+			Self::try_state_members()?;
+			Self::try_state_index()?;
+
+			Ok(())
+		}
+
+		/// ### Invariants of Member storage items
+		///
+		/// Total number of [`Members`] in storage should be >= [`MemberIndex`] of a [`Rank`] in
+		///    [`MemberCount`].
+		/// [`Rank`] in Members should be in [`MemberCount`]
+		fn try_state_members() -> Result<(), sp_runtime::TryRuntimeError> {
+			MemberCount::<T, I>::iter().try_for_each(|(_, member_index)| -> DispatchResult {
+				let total_members = Members::<T, I>::iter().count();
+				ensure!(
+				total_members as u32 >= member_index,
+				"Total count of `Members` should be greater than or equal to the number of `MemberIndex` of a particular `Rank` in `MemberCount`."
+				);
+
+				Ok(())
+			})?;
+
+			Members::<T, I>::iter().try_for_each(|(_, member_record)| -> DispatchResult {
+				ensure!(
+					Self::is_rank_in_member_count(member_record.rank.clone().into()) == true,
+					"`Rank` in Members should be in `MemberCount`"
+				);
+
+				Ok(())
+			})?;
```

Comment:
This isn't a complete invariant for these two storage items. Please check that each MemberCount value is actually consistent with the number of Members at each rank.

Reply:
So, each MemberCount value won't be consistent with the number of Members at each rank because of this statement here https://github.com/paritytech/polkadot-sdk/pull/3007#discussion_r1479457449

MemberCount holds indexes that keeps increasing and Member will always map an account to the rank that account belongs to at a particular time

Reply:
Is it still possible to derive what MemberCount should be based on all the members? Or do I misunderstand

Reply:
Oh, yes. I have added the invariant for this. You can check again now.